digid_eherkenning/saml2/eherkenning.py

```diff
@@ -32,6 +35,214 @@
+                    {
+                        "set_number": "1",
+                        "name": "urn:etoegang:1.9:EntityConcernedID:Pseudo",
+                    },
+                ],
+                "requestedAttributes": [
+                    {
+                        "name": "urn:etoegang:DV:%(oin)s:services:%(index)s"
+                        % {
+                            "oin": eherkenning_config.oin,
+                            "index": eherkenning_config.eidas_attribute_consuming_service_index,
+                        },
+                        "isRequired": False,
+                    }
+                ],
+                "privacy_policy_url": {
+                    "nl": eherkenning_config.privacy_policy,
+                },
+                "herkenningsmakelaars_id": eherkenning_config.makelaar_id,
+                "classifiers": ["eIDAS-inbound"],
+            },
+        ],
+    }
+
+    if eherkenning_config.no_eidas:
+        settings["services"] = settings["services"][:-1]
+
+    return create_service_catalogus(settings)
+
+
+def generate_eherkenning_metadata(eherkenning_config: EherkenningMetadataConfiguration):
+
+    key_file = eherkenning_config.certificate.private_key.path
+    cert_file = eherkenning_config.certificate.public_certificate.path
+
+    setting_dict = {
+        "strict": True,
+        "security": {
+            "signMetadata": True,
+            "authnRequestsSigned": True,
+            "wantAssertionsEncrypted": eherkenning_config.want_assertions_encrypted,
+            "wantAssertionsSigned": eherkenning_config.want_assertions_signed,
+            "soapClientKey": key_file,
+            "soapClientCert": cert_file,
+            "soapClientPassphrase": eherkenning_config.key_passphrase,
+            "signatureAlgorithm": eherkenning_config.signature_algorithm,
+            "digestAlgorithm": eherkenning_config.digest_algorithm,
+            # See comment in the python3-saml for in  OneLogin_Saml2_Response.validate_num_assertions (onelogin/saml2/response.py)
+            # for why we need this option.
+            "disableSignatureWrappingProtection": True,
+            # For eHerkenning, if the Metadata file expires, we sent them an update. So
+            # there is no need for an expiry date.
+            "metadataValidUntil": "",
+            "metadataCacheDuration": "",
+            "requestedAuthnContextComparison": "minimum",
+            "requestedAuthnContext": [eherkenning_config.loa],
+        },
+        # Service Provider Data that we are deploying.
+        "sp": {
+            # Identifier of the SP entity  (must be a URI)
+            "entityId": eherkenning_config.entity_id,
+            # Specifies info about where and how the <AuthnResponse> message MUST be
+            # returned to the requester, in this case our SP.
+            "assertionConsumerService": {
+                "url": furl(
+                    eherkenning_config.base_url + reverse("eherkenning:acs")
```

Comment:
this can probably be done better (test with/without trailing slash on base_url and a subpath on the base url)

Reply:
#19